### PR TITLE
ARXIVNG-2787 Implement mock compiler service.

### PR DESCRIPTION
**Pull request by DavidLFielding**
This PR implements a basic mock compiler service for the submission UI [ARXIVNG-2787].

The compiler service's converter image (performs TeX compilation) is very large and takes extra time and disk space to download and install. The mock compiler allows developers to skip installation of the large convert image. This is mainly useful when you are evaluating the submission UI and not interested in the compilation details for a specific TeX-based paper.

The initial mock compiler service responds to compilation service API requests and provides a fixed compilation log (/log) and PDF (/product).

By substituting your own PDF and autotex.log file (as compilation_log), and rebuilding the mock compiler service,  you may simulate the visual behavior of a specific TeX-based paper. This might be useful when investigating log highlighting on the process page.

I've refactored and reworked the README.md file to have two threads, one for the mock compiler service, and one for the full compilation service. Please feel free to make improvements to the documentation.

To make installation and configuration of the mock compilation service easier for developers I've created a separate docker-compose configuration file call "docker-compose-mock-compiler.yml". Use this configuration file with the '-f <config>' argument to all docker-compose commands.

The mock compiler service is intended to look and behave like the real thing. The process page checks the compilation status when you advance to the page so the mock compile service must transition through different states. Once you initiate the compilation the mock compile returns the 'In Progress" status in order to render the "Processing Underway" view. The state then changes to 'Complete" for the next status check (after five seconds page refreshes). At this point the PDF and log are available in the process page.

The idea is that someone using the submission UI with mock compiler shouldn't notice anything different from the live compilation service other than the PDF and log being static. If you use the upload2.tar.gz test submission (File Manager Tests) you may not even realize you are using the mock compiler service.

Clearly, there are a lot of things we could do with the mock service, like making it easier to substitute in PDF/logs, and support this for a series of submissions. The initial version seems workable for now.

You should be able to check out the branch, build with "-f docker-compose-mock-compiler.yml" added to all docker-compose commands, and bring up the mock compiler service. Alternatively, you could overwrite the default configuration file with the mock configuration file.

Note: When using the mock compiler service the compiler-api and compiler-worker do not show up in a 'docker-compose ps' command. You should check to make sure the correct series of micoservices is running, including a 'mock-compiler-api' service.

[ARXIVNG-2787]: https://arxiv-org.atlassian.net/browse/ARXIVNG-2787

### CODE REVIEW

File: mock-services/data/compiler/README [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Data for mock compiler service.
```
Comment (mhl10) [inline mock-services/data/compiler/README]: Minor: rename file from README to README.md even though it currently doesn't use any markup.

File: README.md [hunk 2]
```diff
@@ -35,20 +127,9 @@ To test if you have access to ECR run:
...
-to work properly.
-
-You may set these to your real AWS credentials or simply set them to arbitrary
-dummy values.
-
-```bash
-export AWS_ACCESS_KEY_ID=fookey
-export AWS_SECRET_ACCESS_KEY=foosecretkey
-```
+Make sure you have your AWS settings configured properly before proceeding (see above).
```
Comment (mhl10) [inline README.md]:
```suggestion
Make sure you have your AWS settings configured properly before proceeding ([see above](#aws-credentials)).
```

File: README.md [hunk 3]
```diff
@@ -114,6 +197,12 @@ use the ``-v`` flag to drop old volumes.
 docker-compose rm -v
 ```
 
+Add the '-f' flag when using the mock compiler service:
+
+```bash
+docker-compose -f docker-compose-mock-compiler.yml rm -v
```
Comment (mhl10) [inline README.md:207]: Is that meant to remove the mock compiler service?

**Review by mhl10 — APPROVED** [review]
Looks good to me, appreciate the clear documentation! I added some very minor comments.

File: mock-services/data/compiler/README (hunk 1 above)
Comment (DavidLFielding) [reply mock-services/data/compiler/README]: Done.

File: README.md (hunk 3 above)
Comment (DavidLFielding) [reply README.md:207]: It is my belief that many docker-compose commands rely on the config file when deciding what to do.

When I originally ran the basic cleanup command it looked like mock compiler was not getting cleaned up properly. It does not show up in 'docker-compose ps' output but it appears to linger in the 'docker ps -a' listing with an exit status.

lib-dlf2-maclap:arxiv-submission-ui dlf2$ docker ps -a
CONTAINER ID        IMAGE                 COMMAND                  CREATED             STATUS                            PORTS               NAMES
4793feb67df9        arxiv/mock-compiler   "pipenv run uwsgi --…"   41 hours ago        Exited (137) About a minute ago                       mock-compiler-api

When I add the -f argument this lingering entry gets removed.

lib-dlf2-maclap:arxiv-submission-ui dlf2$ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

I believe the config file might direct the cleanup process, and thus the mock compiler doesn't seem to get properlly cleaned up when using the non-mock default config.

I'm not sure if this matters all that much so I can remove this if you like. I just don't like the entry hanging around when the cleanup command cleaned out the rest. 

There is clearly a difference between 'docker-compose ps -a' output and 'docker ps -a'.

Comment (mhl10) [reply README.md:207]: Sounds good, might be good to update the README.md with a little explanation, e.g.

Add the '-f' flag when using the mock compiler service  to remove any lingering mock compiler image:

**Review by SBBCornell — COMMENTED** [review]
I dont think I know how to review this PR, sorry. Appending "-f docker-compose-mock-compiler.yml" to the docker commands does not spin up submission-ui, just displays information about the docker command. Spinning up this branch without appending it appears to work as expected, though Im not sure what to look for...

File: README.md (hunk 3 above)
Comment (DavidLFielding) [reply README.md:207]: Done

**Comment by SBBCornell** [comment]
> I dont think I know how to review this PR, sorry. Appending "-f docker-compose-mock-compiler.yml" to the docker commands does not spin up submission-ui, just displays information about the docker command. Spinning up this branch without appending it appears to work as expected, though Im not sure what to look for...

David straightened me out on this. There is no error, I was typing the commands incorrectly.

**Review by bdc34 — APPROVED** [review]
Looks great.

**Pull request merged**